library/sha512.c

```diff
@@ -50,12 +50,128 @@
 #endif /* MBEDTLS_PLATFORM_C */
 #endif /* MBEDTLS_SELF_TEST */
 
+#if defined(__aarch64__)
+#  if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT) || \
+      defined(MBEDTLS_SHA512_USE_A64_CRYPTO_ONLY)
+#    include <arm_neon.h>
+#  endif
+#  if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT)
+#    if defined(__unix__)
+#      if defined(__linux__)
+         /* Our preferred method of detection is getauxval() */
+#        include <sys/auxv.h>
+#      endif
+       /* Use SIGILL on Unix, and fall back to it on Linux */
+#      include <signal.h>
+#    endif
+#  endif
+#elif defined(_M_ARM64)
+#  if defined(MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT) || \
```

Comment:
We know that the sha384 is one special case of sha512, will this patch implement it?

Reply:
As with SHA-224 and the SHA-256 PR, SHA-384 is separate to this PR.

To enable SHA-384 in Mbed TLS, it is required to enable SHA-512. If this acceleration is also enabled, then both SHA-512 and SHA-384 will benefit from it.

Reply:
Ok, got it. Seems there will be one seperate PR for sha384, that make sense.

Reply:
Aha. The optimization focus on the update part. maybe I should check the init for the initial vector and final for the final selection of hash values for sha512 and sha384 seperately.